From 2a08d7bd3e8244224b5cb64e08802125adf53fbc Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Thu, 15 Jun 2017 15:39:50 -0500
Subject: [PATCH 1902/5242] MLK-15076: arm: dts: insulate ULP kernel boot from
 M4 boot

commit  ec0298439dd81e2698f4228bdc8a3ce0d9e699b7 from
https://source.codeaurora.org/external/imx/linux-imx.git

i.MX7ULP QSPI dtb was used to update the M4 images, it should not able
to boot the kernel even without the M4 image in QSPI.

Also fixed the typo in dtsi to correct the QSPI register address
mapping range.

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk-qspi.dts |   30 ++++++++++++++++++++++++++++++
 arch/arm/boot/dts/imx7ulp.dtsi         |    2 +-
 2 files changed, 31 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx7ulp-evk-qspi.dts b/arch/arm/boot/dts/imx7ulp-evk-qspi.dts
index bc9db44..e760588 100644
--- a/arch/arm/boot/dts/imx7ulp-evk-qspi.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk-qspi.dts
@@ -9,6 +9,18 @@
 
 #include "imx7ulp-evk.dts"
 
+/ {
+	regulators {
+		compatible = "simple-bus";
+
+		dummy: regulator-dummy {
+		compatible = "regulator-fixed";
+		regulator-name = "dummy";
+		regulator-always-on;
+		};
+	};
+};
+
 &qspi1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_qspi1_1>;
@@ -23,6 +35,24 @@
 	};
 };
 
+&rpmsg{
+	status = "disabled";
+};
+
+&cpu0 {
+	arm-supply= <&dummy>;
+	operating-points = <
+		/* KHz  uV */
+		503666  1025000
+	>;
+};
+
+&usdhc0 {
+	vqmmc-supply = <&dummy>;
+	no-1-8-v;
+	non-removable;
+};
+
 &iomuxc {
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx7ulp.dtsi b/arch/arm/boot/dts/imx7ulp.dtsi
index a8ebe44..c8ca3e9 100644
--- a/arch/arm/boot/dts/imx7ulp.dtsi
+++ b/arch/arm/boot/dts/imx7ulp.dtsi
@@ -579,7 +579,7 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			compatible = "fsl,imx7ulp-qspi";
-			reg = <0x410A5000 0x10000>, <0xC0000000 0x10000000>;
+			reg = <0x410A5000 0x1000>, <0xC0000000 0x10000000>;
 			reg-names = "QuadSPI", "QuadSPI-memory";
 			interrupts = <GIC_SPI 64 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&clks IMX7ULP_CLK_DUMMY>,
-- 
1.7.9.5

